lib/trino-collect/src/main/java/io/trino/collect/cache/EvictableCache.java

```diff
@@ -125,18 +124,22 @@ public V get(K key)
     {
         Token<K> newToken = new Token<>(key);
         Token<K> token = tokens.computeIfAbsent(key, ignored -> newToken);
-        try {
-            return dataCache.get(token);
-        }
-        catch (Throwable e) {
-            if (newToken == token) {
+        if (token == newToken) {
+            try {
+                return dataCache.get(newToken);
+            }
+            catch (Throwable e) {
```

Comment:
Should we apply this same approach to `get(K key, Callable<? extends V> valueLoader)` ?

Reply:
Nevermind, wrong assumption.